lib/ansible/cli/inventory.py

```diff
@@ -83,11 +87,20 @@ def parse(self):
         self.parser.add_option_group(action_group)
 
         # Options
+
+        # graph
         self.parser.add_option("-y", "--yaml", action="store_true", default=False, dest='yaml',
                                help='Use YAML format instead of default JSON, ignored for --graph')
         self.parser.add_option("--vars", action="store_true", default=False, dest='show_vars',
                                help='Add vars to graph display, ignored unless used with --graph')
 
+        # list
+        self.parser.add_option("--export", action="store_true", default=C.INVENTORY_EXPORT, dest='export',
+                               help="When doing an --list, represent in a way that is optimized for export,"
+                                    "not as an accurate representation of how Ansible has processed it")
```

Comment:
The help blurb is kind of vague. optimized how and for what?

Reply:
for export .. as for 'how' .. it seems too long to detail here.

Reply:
Just to make an attempt at word-smithing

> When doing an --list, represent shared variables inside groups or the inventory,
> which has a smaller memory footprint but is not Ansible's internal representation

Reply:
i'll just merge as is for now, let the 'wordsmithing' update it in latter patches so as to not keep this waiting